core/src/test/java/org/apache/calcite/rex/RexBuilderTest.java

```diff
@@ -957,6 +957,16 @@ private void checkDate(RexLiteral literal) {
     assertThat(inCall.getKind(), is(SqlKind.SEARCH));
   }
 
+  @Test void testMakeInSingleValue() {
+    RelDataTypeFactory typeFactory = new SqlTypeFactoryImpl(RelDataTypeSystem.DEFAULT);
+    RexBuilder rexBuilder = new RexBuilder(typeFactory);
+    RelDataType intType = typeFactory.createSqlType(SqlTypeName.INTEGER);
+    RexNode column = rexBuilder.makeInputRef(intType, 0);
+    RexLiteral literal = rexBuilder.makeLiteral(100, intType);
+    RexNode inCall = rexBuilder.makeIn(column, ImmutableList.of(literal));
```

Comment:
Note that `makeIn(column, ImmutableList.of(literal, literal))` currently returns a singleton Sarg but could be optimized.

Reply:
Modified the current approach in https://github.com/apache/calcite/pull/3991/commits/32f0485ff2225f48d312566930bb12df384ec7f1 to account for duplicates. 

I also added more tests for this purpose in https://github.com/apache/calcite/pull/3991/commits/a4dbffdd04ae802ae53687de54a9e01e762af640.